ql/src/java/org/apache/hadoop/hive/ql/scheduled/ScheduledQueryExecutionService.java

```diff
@@ -116,6 +133,37 @@ private void sleep(long idleSleepTime) throws InterruptedException {
       }
     }
 
+  }
+
+  private void workerStarted(ScheduledQueryExecutor executor) {
+    runningWorkers.add(executor);
+    usedExecutors.incrementAndGet();
+  }
+
+  private void workerStopped(ScheduledQueryExecutor executor) {
+    usedExecutors.decrementAndGet();
+    runningWorkers.remove(executor);
+    forceScheduleCheck();
+  }
+
+  class ScheduledQueryExecutor implements Runnable {
```

Comment:
Similar to `ScheduledQueryPoller` , can we leave a description comment for `ScheduledQueryExecutor` too (even if evident)? Thanks

Reply:
also renamed a few methods to use the "executor" word more consistently